pinot-core/src/main/java/org/apache/pinot/core/operator/transform/transformer/datetime/EpochToEpochTransformer.java

```diff
@@ -29,14 +31,20 @@
 public class EpochToEpochTransformer extends BaseDateTimeTransformer<long[], long[]> {
 
   public EpochToEpochTransformer(DateTimeFormatSpec inputFormat, DateTimeFormatSpec outputFormat,
-      DateTimeGranularitySpec outputGranularity) {
-    super(inputFormat, outputFormat, outputGranularity);
+      DateTimeGranularitySpec outputGranularity, @Nullable DateTimeZone bucketingTz) {
+    super(inputFormat, outputFormat, outputGranularity, bucketingTz);
   }
 
   @Override
   public void transform(@Nonnull long[] input, @Nonnull long[] output, int length) {
-    for (int i = 0; i < length; i++) {
-      output[i] = transformMillisToEpoch(transformToOutputGranularity(transformEpochToMillis(input[i])));
+    if (useCustomBucketingTimeZone()) {
```

Comment:
We can still short-circuit it if it is UTC. Same for other places.
I guess we can check if the explicitly specified time zone is the same as output format time zone, and count it as custom time zone only when they are different.

Reply:
If time zone is set (even if utc) I'd rather stick to the more intuitive datatime truncation approach than to revert to unintuitive one.

Reply:
Will it add extra overhead when extra bucketing is not needed?

Reply:
That conditional is rarely invoked and checks a final variable so it's probably optimized away. Benchmark results are practically the same.

Benchmark               (_numRows)                                                                                                                       (_query)  (_scenario)  Mode  Cnt    Score    Error  Units
BenchmarkQueries.query    15000000  SELECT * FROM MyTable WHERE  dateTimeConvert(TSTMP_COL, '1:MILLISECONDS:EPOCH', '1:MILLISECONDS:EPOCH', '1:DAYS') = 120000000   EXP(0.001)  avgt    5  344.870 ± 20.175  ms/op
BenchmarkQueries.query    15000000  SELECT * FROM MyTable WHERE  dateTimeConvert(TSTMP_COL, '1:MILLISECONDS:EPOCH', '1:MILLISECONDS:EPOCH', '1:DAYS') = 120000000     EXP(0.5)  avgt    5  355.147 ± 39.399  ms/op
BenchmarkQueries.query    15000000  SELECT * FROM MyTable WHERE  dateTimeConvert(TSTMP_COL, '1:MILLISECONDS:EPOCH', '1:MILLISECONDS:EPOCH', '1:DAYS') = 120000000   EXP(0.999)  avgt    5  342.480 ± 11.843  ms/op

vs

Benchmark               (_numRows)                                                                                                                       (_query)  (_scenario)  Mode  Cnt    Score    Error  Units
BenchmarkQueries.query    15000000  SELECT * FROM MyTable WHERE  dateTimeConvert(TSTMP_COL, '1:MILLISECONDS:EPOCH', '1:MILLISECONDS:EPOCH', '1:DAYS') = 120000000   EXP(0.001)  avgt    5  347.397 ± 37.452  ms/op
BenchmarkQueries.query    15000000  SELECT * FROM MyTable WHERE  dateTimeConvert(TSTMP_COL, '1:MILLISECONDS:EPOCH', '1:MILLISECONDS:EPOCH', '1:DAYS') = 120000000     EXP(0.5)  avgt    5  347.330 ± 18.109  ms/op
BenchmarkQueries.query    15000000  SELECT * FROM MyTable WHERE  dateTimeConvert(TSTMP_COL, '1:MILLISECONDS:EPOCH', '1:MILLISECONDS:EPOCH', '1:DAYS') = 120000000   EXP(0.999)  avgt    5  347.131 ± 53.469  ms/op